src/Loader.php

```diff
@@ -667,10 +667,20 @@ public static function add_admin_body_classes( $admin_body_class = '' ) {
 			$classes[] = 'woocommerce-embed-page';
 		}
 
-		// Some routes or features like onboarding hide the wp-admin navigation and masterbar. Setting `woocommerce_admin_is_loading` to true allows us
-		// to premeptively hide these elements while the JS app loads. This class is removed when `<Layout />` is rendered.
-		if ( self::is_admin_page() && apply_filters( 'woocommerce_admin_is_loading', false ) ) {
+		/**
+		 * Some routes or features like onboarding hide the wp-admin navigation and masterbar.
+		 * Setting `woocommerce_admin_is_loading` to true allows us to premeptively hide these
+		 * elements while the JS app loads. This class is removed when `<Layout />` is rendered.
+		 *
+		 * @param bool $is_loading If WooCommerce Admin is loading a fullscreen view.
+		 */
+		$is_loading = apply_filters( 'woocommerce_admin_is_loading', false );
+
+		if ( self::is_admin_page() && $is_loading ) {
 			$classes[] = 'woocommerce-admin-is-loading';
+			// Include the full screen class now to prevent the dashboard
+			// from being shown while the <ProfileWizard /> chunk loads.
+			$classes[] = 'woocommerce-admin-full-screen';
```

Comment:
I'm a little hesitant to add the `woocommerce-admin-full-screen` class to every loading page.  It contains styles that are specific to full-screen pages and creates one more class that needs to be removed for components making use of the `woocommerce_admin_is_loading` filter.

Reply:
Removed in 8160ba3 in favor of delaying the removal of the loading class.